src/instrumenter/state_var_instrumenter.ts

```diff
@@ -842,7 +840,38 @@ export function interposeTupleAssignment(
         marker = temporaryUpdateStmt;
         ctx.addGeneralInstrumentation(temporaryUpdateStmt);
 
+        res.push([temporaryUpdate, freshLHSToPathM.get(temporary.id) as number[]]);
+    }
+
+    return res;
+}
+
+/**
+ * Complex interposing case - when `updateNode` is a (potentially nested) tuple assignment with multiple
+ * state vars and other expressions updated together.
+ *
+ * The strategy here is to replace all expressions inside tuples on the LHS of `updateNode` with temporaries, then assign
+ * those temporaries one-by-one to the original LHS expressions (while interposing with wrappers wherever we have properties to check).
+ *
+ * Special care needs to be taken for any indexing sub-expressions appearing inside the LHS expressions - those need to be computed before
+ * the assignment to preserve evaluation order.
+ *
+ * @param ctx
+ * @param updateNode
```

Comment:
We can skip these, as description is empty.

Reply:
Fixed in b4b6be9.